src/main/java/org/spongepowered/api/event/SpongeEventFactory.java

```diff
@@ -189,6 +197,25 @@
     private SpongeEventFactory() {
     }
 
+    private static Class<?> getBaseClass(Class<?> event) {
+        Class<?> superClass = null;
+        for (EventFactoryPlugin plugin: plugins) {
+            superClass = plugin.resolveSuperClassFor(event, superClass);
+        }
+        return superClass;
```

Comment:
There should always be a superclass returned, either through a custom plugin, or by `AnnotationEventFactoryPlugin` reading the `@ImplementedBy` annotation from `Event`. If null *is* returned, it's most likely an error, and should be exposed as such.